metadata/src/lib.rs

```diff
@@ -8,6 +8,8 @@ use frame_metadata::{
 use scale_info::{form::PortableForm, Field, PortableRegistry, TypeDef, Variant};
 use std::collections::HashSet;
 
+pub mod retain;
```

Comment:
let's just expose the single function we need from the module rather than the module itself :)